### PR TITLE
Run make docs step in gitlab-ci.yml

### LINKED ISSUE #3431: Run make docs step in gitlab-ci.yml.
This should ensure that docs don't break.

┆Issue is synchronized with this [Jira Task](https://ucsc-cgl.atlassian.net/browse/TOIL-785)
┆Issue Number: TOIL-785

### PR DESCRIPTION
**Pull request by jonathanxu18**
Closes #3431.

This PR adds a step in gitlab-ci.yml that runs make docs to ensure that the docs don't break.

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -33,11 +33,18 @@ after_script:
   - stopdocker || true
 
 stages:
+  - docs
```
Comment (mr-c) [inline .gitlab-ci.yml:36]: Can this be in parallel with the `linting` step?

Comment (jonathanxu18) [reply .gitlab-ci.yml]: Yep, just added it in. Thanks for the suggestion!

**Review by DailyDreaming — COMMENTED** [review]
`make docs` is generating files that mypy doesn't like.  You can either run `make docs` after `make mypy` or run `make clean_docs`.

**Review by DailyDreaming — APPROVED** [review]
LGTM.

**Pull request merged**